### PR TITLE
Create login dialog for upsell pages

### LINKED ISSUE #7680: [North Star Applicant UI] Create all remaining login prompt modals
We have already created a modal dialog for a login prompt that is shown when an applicant clicks on the program in the program index page.

Remaining login prompts:
1) Dialog shown on summary/review page when redirected from program slug. Replaces [this code](https://github.com/civiform/civiform/blob/main/server/app/controllers/applicant/ApplicantProgramReviewController.java#L158). We will need to add the ability to show the modal on load, as well as create a second repeat open group.
1) Upsell/application confirmation page (basic dialog)
1) Upsell/application confirmation page for common intake (same as the standard confirmation page)

### PR DESCRIPTION
**Pull request by greg-hyde**
### Description

When the user is a guest, show a login dialog as an interstitial in the upsell pages when the user goes to apply to another program

#### General

Read the full guidelines for PRs [here](https://github.com/civiform/civiform/wiki/Technical-contribution-guide#creating-a-pull-request)

- [x] Added the correct label: < feature | enhancement | bug | under-development | dependencies | infrastructure | ignore-for-release | database >
- [x] Assigned to a specific person, `civiform/developers`, or a [more specific round-robin list](https://github.com/civiform/civiform/wiki/Technical-contribution-guide#adding-reviewers)
- [x] Added an additional reviewer from outside your organization as FYI (if the primary reviewer is in the same organization as you)
- [x] Created unit and/or browser tests which fail without the change (if possible)
- [x] Performed manual testing (Chrome and Firefox if it includes front-end changes)

#### User visible changes

- [x] Followed steps to [internationalize](https://github.com/civiform/civiform/wiki/Internationalization-%28i18n%29#application-strings) any new strings
  - [x] Added context strings to new [messages](https://github.com/civiform/civiform/blob/main/server/conf/i18n/messages)
  - [x] Didn't use a message in applicant facing code that isn't translated yet (unless behind a flag)
- [x] Wrote browser tests using the [validateAccessibility](https://sourcegraph.com/github.com/civiform/civiform/-/blob/browser-test/src/support/index.ts?L437:14&subtree=true) method
- [x] Tested on mobile view. See [mobile device mode](https://developer.chrome.com/docs/devtools/device-mode/)
- [x] Manually tested at 200% size
- [x] Manually evaluated tab order

#### New Features

- [x] Conditioned new functionality on a [FeatureFlag](https://github.com/civiform/civiform/wiki/Feature-Flags)
- [x] Wrote browser tests with the feature flag off and on, etc.

### Issue(s) this completes

Fixes #7680

### CODE REVIEW

**Comment by greg-hyde** [comment]
* **#7876** <a href="https://app.graphite.dev/github/pr/civiform/civiform/7876?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @greg-hyde and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: browser-test/src/applicant/northstar_upsell.test.ts [hunk 1]
```diff
@@ -20,32 +20,66 @@ test.describe('Upsell tests', {tag: ['@northstar']}, () => {
...
+    await enableFeatureFlag(page, 'north_star_applicant_ui')
+
+    await test.step('Submit application', async () => {
+      await applicantQuestions.clickApplyProgramButton(programName)
+      await applicantQuestions.submitFromReviewPage(
+        /* northStarEnabled= */ true,
+      )
+    })
+
+    await test.step('Validate screenshot and accessibility', async () => {
```
Comment (ssandbekkhaug) [inline browser-test/src/applicant/northstar_upsell.test.ts:73]: Optional Nit: delete this test.step? It's enclosing another one

File: server/app/views/applicant/ModalFragment.html [hunk 2]
```diff
@@ -1,5 +1,5 @@
 <div
-  th:fragment="modalContainer(showProgramIndexLogin, showReviewLogin)"
+  th:fragment="modalContainer(showProgramIndexLogin, showReviewLogin, showUpsellLogin)"
```
Comment (ssandbekkhaug) [inline server/app/views/applicant/ModalFragment.html:2]: Why is this a mega-div? Could it be separated into 3 fragments so that callers just have to call:

~{applicant/ModalFragment:: programIndexLogin)

instead of:

~{applicant/ModalFragment:: modalContainer(true,
                                                             false,
                                                             false)}

?

File: server/app/views/applicant/IneligibleTemplate.html [hunk 3]
```diff
@@ -52,7 +52,8 @@
     </div>
     <div
       th:replace="~{applicant/ModalFragment:: modalContainer(<!--/* showProgramIndexLogin= */-->false,
```
Comment (ssandbekkhaug) [inline server/app/views/applicant/IneligibleTemplate.html:54]: If all params are false, why is this needed?

File: server/app/views/applicant/IneligibleTemplate.html [hunk 4]
```diff
@@ -52,7 +52,8 @@
...
-                                                             <!--/* showReviewLogin= */-->false)}"
+                                                             <!--/* showReviewLogin= */-->false,
```
Comment (ssandbekkhaug) [inline server/app/views/applicant/IneligibleTemplate.html:55]: If all params are false, why is this needed?

Comment (greg-hyde) [reply server/app/views/applicant/IneligibleTemplate.html:55]: we still show the devtools modal in all pages

File: server/app/views/applicant/ModalFragment.html (hunk 2 above)
Comment (greg-hyde) [reply server/app/views/applicant/ModalFragment.html:2]: it is mostly because of the dev tools modal which is shown on all pages. I'm open to updating it though. Thus far everything has been contained in the #modal-container element to match the previous implementation, but that's not really necessary

File: browser-test/src/applicant/northstar_upsell.test.ts [hunk 5]
```diff
@@ -20,32 +20,64 @@ test.describe('Upsell tests', {tag: ['@northstar']}, () => {
...
+    })
+
+    await test.step('Validate that login dialog is shown when user clicks on apply to another program', async () => {
+      await applicantQuestions.clickApplyToAnotherProgramButton()
+      await expect(page.getByTestId('login')).toContainText(
+        'Create an account or sign in',
+      )
+      await validateScreenshot(page, 'upsell-north-star-login')
+
+      await validateAccessibility(page)
```
Comment (rockycodes) [inline browser-test/src/applicant/northstar_upsell.test.ts:80]: suggestion (non-blocking): I just tried this out on Firefox and for some reason it doesn't work to tab to the "Log in" button on the modal. I'm able to tab to this button in Chrome, and in Firefox when Voiceover is on, but not with Voiceover off. Would you make an issue to look into this? This could be a Firefox bug and maybe not a problem, but we are able to tab to all the buttons on our current modal so I'm curious what's different. On the plus side, tabbing through the elements never takes you behind the modal which is great!

**Review by rockycodes — COMMENTED** [review]
reviewed!

File: server/app/views/applicant/ModalFragment.html (hunk 2 above)
Comment (greg-hyde) [reply server/app/views/applicant/ModalFragment.html:2]: ok now that I tried the refactoring, I remember why I originally did it this way. modal.ts (the non north star version) asserts that there is a modal container element containing all modals, and we can't get rid of that until after north star is launched.

So I did the refactoring and I like it better, but I think it has to wait until after north star launch when we can remove modal.ts. Filed an issue here https://github.com/civiform/civiform/issues/7880 with a draft PR linked

File: browser-test/src/applicant/northstar_upsell.test.ts (hunk 5 above)
Comment (greg-hyde) [reply browser-test/src/applicant/northstar_upsell.test.ts:80]: that's weird I can tab through it just fine on firefox...

Comment (rockycodes) [reply browser-test/src/applicant/northstar_upsell.test.ts:80]: weird! sounds like it's something on my end then.

**Review by ssandbekkhaug — APPROVED** [review]
LGTM
